instrumentation/jaxrs2/src/main/java/brave/jaxrs2/TracingContainerFilter.java

```diff
@@ -0,0 +1,147 @@
+package brave.jaxrs2;
+
+import brave.ServerHandler;
+import brave.Span;
+import brave.Tracer;
+import brave.Tracer.SpanInScope;
+import brave.Tracing;
+import brave.http.HttpServerHandler;
+import brave.http.HttpServerParser;
+import brave.internal.Nullable;
+import brave.propagation.TraceContext;
+import java.lang.annotation.Annotation;
+import javax.annotation.Priority;
+import javax.ws.rs.ConstrainedTo;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+import zipkin.Constants;
+
+import static javax.ws.rs.RuntimeType.SERVER;
+
+@Provider
+@Priority(0) // to make the span in scope visible to other filters
+@ConstrainedTo(SERVER)
```

Comment:
should we add https://jersey.java.net/apidocs/2.12/jersey/javax/ws/rs/container/PreMatching.html on here as well so this provider happens before any routing decisions are made?